javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java

```diff
@@ -109,6 +110,42 @@ private TreeUtils() {
   /** The value of Flags.COMPACT_RECORD_CONSTRUCTOR which does not exist in Java 9 or 11. */
   static final long Flags_COMPACT_RECORD_CONSTRUCTOR = 1L << 51;
 
+  // These variables cannot be final because they might be overwritten in the static block
+  // immediately below.
+  /** The {@code CaseTree.getExpressions()} method. Null on JDK 11 and lower. */
+  private static @Nullable Method caseGetExpressions = null;
+  /** The {@code CaseTree.getBody()} method. Null on JDK 11 and lower. */
+  private static @MonotonicNonNull Method caseGetBody = null;
+  /** The {@code BindingPatternTree.getVariable()} method. Null on JDK 11 and lower. */
+  private static @MonotonicNonNull Method bindingPatternGetVariable = null;
+  /** The {@code InstanceOfTree.getPattern()} method. Null on JDK 11 and lower. */
+  private static @MonotonicNonNull Method instanceOfGetPattern = null;
+  /** The {@code SwitchExpressionTree.getExpression()} method. Null on JDK 11 and lower. */
+  private static @MonotonicNonNull Method switchExpressionGetExpression = null;
+  /** The {@code SwitchExpressionTree.getCases()} method. Null on JDK 11 and lower. */
+  private static @MonotonicNonNull Method switchExpressionGetCases = null;
+  /** The {@code YieldTree.getValue()} method. Null on JDK 11 and lower. */
+  private static @MonotonicNonNull Method yieldGetValue = null;
+
+  static {
+    if (SystemUtil.jreVersion >= 12) {
+      try {
+        caseGetExpressions = CaseTree.class.getDeclaredMethod("getExpressions");
+        caseGetBody = CaseTree.class.getDeclaredMethod("getBody");
+        Class<?> bindingPatternClass = Class.forName("com.sun.source.tree.BindingPatternTree");
+        bindingPatternGetVariable = bindingPatternClass.getMethod("getVariable");
+        instanceOfGetPattern = InstanceOfTree.class.getMethod("getPattern");
+        Class<?> switchExpressionClass = Class.forName("com.sun.source.tree.SwitchExpressionTree");
+        switchExpressionGetExpression = switchExpressionClass.getMethod("getExpression");
+        switchExpressionGetCases = switchExpressionClass.getMethod("getCases");
+        Class<?> yieldTreeClass = Class.forName("com.sun.source.tree.YieldTree");
+        yieldGetValue = yieldTreeClass.getMethod("getValue");
+      } catch (ClassNotFoundException | NoSuchMethodException e) {
+        throw new BugInCF("JDK 17+ reflection problem", e);
```

Comment:
This exception message is confusing, because (given the test above against JDK version 12), a user could get this message when running on a JDK whose version is in the range 12-16. I know we don't **technically** support those versions, but I think the error message should reflect the actual conditions under which this error might occur.